src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIRLoweringPhase.cs

```diff
@@ -483,7 +483,7 @@ public override void VisitTagHelperBlock(TagHelperChunkGenerator chunkGenerator,
                     tagName = tagName.Substring(_tagHelperPrefix.Length);
                 }
 
-                _builder.Push(new InitializeTagHelperStructureIRNode()
+                _builder.Push(new TagHelperBodyIRNode()
                 {
                     TagName = tagName,
```

Comment:
Conceptually it doesn't make a lot of sense to have tag name/tag mode on the body. Lets move these to `TagHelperIRNode`